js/jquery.mapael.js

```diff
@@ -380,6 +380,14 @@
 					Mapael.updateElem(elemOptions, links[id], $tooltip, animDuration);
 				}
 				
+				// Update legends
+				$.merge(legends, Mapael.createLegends($self, options, "area", areas, 1));
```

Comment:
I think that the line 384 should be replaced by : 

legends = Mapael.createLegends($self, options, "area", areas, 1);

In order to reset the legends collection because we re-create all the legends from scratch, otherwise the collection will keep the previous obselete legends.

(By the way, maybe the variable "legends" should be removed later, as it is only used for hiding legend elements. It will not be used anymore when your other PR will be merged : https://github.com/neveldo/jQuery-Mapael/pull/85)